### PR TITLE
Making the project prettier

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
This PR makes changes to the presentation of the module, but not to its functionality. Actions taken:

- [beb2431](https://github.com/EduardGomezEscandell/GoWSL/pull/11/commits/beb24316f2cb274240a0a263424ea2f7f4bbf4a5) Documentation:
  - Fixes typos.
  - Adds module documentation comment.
  - Expands existing documentation comments.
- [af9454b](https://github.com/EduardGomezEscandell/GoWSL/pull/11/commits/af9454bed53daab27628f9b3c6b44123a3d1491d): Adds `nolintlint` and fixes a couple redundant `nolint` comments.
- [5398283](https://github.com/EduardGomezEscandell/GoWSL/pull/11/commits/53982833ae73f1239117b34e11566114b1c03eb5): Updates the README
- [fa3a2c8](https://github.com/EduardGomezEscandell/GoWSL/pull/11/commits/fa3a2c85c06555066a0d2595fd48cd1d166673ee): Adds `.exe` files to the gitignore list.

Do not squash-merge: the commits are rather different between each other.

### CODE REVIEW

File: distro.go [hunk 1]
```diff
@@ -100,6 +107,7 @@ func (d *Distro) DriveMountingEnabled(value bool) error {
 }
 
 // GetConfiguration is a wrapper around Win32's WslGetDistributionConfiguration.
+// It returns a configuration ofbect with information about the distro.
```
Comment (didrocks) [inline distro.go:110]:
```suggestion
// It returns a configuration object with information about the distro.
```

File: distro.go [hunk 2]
```diff
@@ -137,7 +145,7 @@ func (d Distro) GetConfiguration() (c Configuration, e error) {
 	return conf, nil
 }
 
-// String deserializes a Configuration object as a yaml string.
+// String deserializes a distro and its configurantion as a yaml string.
```
Comment (didrocks) [inline distro.go:148]:
```suggestion
// String deserializes a distro and its configuration as a yaml string.
```

File: exec.go [hunk 3]
```diff
@@ -72,7 +74,7 @@ func (m ExitError) Error() string {
 
 // Is ensures ExitErrors can be matched with errors.Is().
 func (m ExitError) Is(target error) bool {
-	_, ok := target.(ExitError) // nolint: errorlint
+	_, ok := target.(ExitError) //nolint: errorlint
```
Comment (didrocks) [inline exec.go:77]: ah, hence the nolinlint addition :)

**Review by didrocks — APPROVED** [review]
2 typos, otherwise all good! :) Feel free to merge once fixed.

**Comment by EduardGomezEscandell** [comment]
> 2 typos, otherwise all good! :) Feel free to merge once fixed.

You're slowly becoming my english spelling teacher 😅. Too bad misspell doesn't catch these, I created an issue just to have it documented somwhere: https://github.com/EduardGomezEscandell/GoWSL/issues/12

**Comment by EduardGomezEscandell** [comment]
Force-pusehd after squashing typo fixes with the relevant commit.

**Comment by didrocks** [comment]
> > 2 typos, otherwise all good! :) Feel free to merge once fixed.
> 
> You're slowly becoming my english spelling teacher sweat_smile. Too bad misspell doesn't catch these, I created an issue just to have it documented somwhere: #12

I would rather likely high level you on the number of spelling someone can write in a single PR TBH :p

**Pull request merged**